test-ts/Slasher.ts

```diff
@@ -0,0 +1,538 @@
+
+
+    before(async function () {
+        [tokenAddresses, orderbook, renExSettlement, renExBalances, rewardVault] = await setup(darknode, slasher);
+        eth_address = tokenAddresses[ETH].address
+        eth_decimals = new BigNumber(10).pow(tokenAddresses[ETH].decimals())
+    });
+
+
+    it("should correctly relocate fees", async () => {
+        const tokens = market(BTC, ETH);
+        const buy = { settlement: 2, tokens, price: 1, volume: 2 /* BTC */, minimumVolume: 1 /* ETH */ };
+        const sell = { settlement: 2, tokens, price: 0.95, volume: 1 /* ETH */ };
+
+        let [btcAmount, ethAmount, buyOrderID, sellOrderID] = await submitMatch(buy, sell, buyer, seller, darknode, renExSettlement, renExBalances, tokenAddresses, orderbook, false);
+        btcAmount.should.eql(0.975 /* BTC */);
+        ethAmount.should.eql(1 /* ETH */);
+
+        let guiltyOrderID = buyOrderID;
+        let guiltyAddress = buyer;
+        let innocentOrderID = sellOrderID;
+        let innocentAddress = seller;
+        (await orderbook.orderMatch(guiltyOrderID))[0].should.eql(innocentOrderID);
+        (await orderbook.orderMatch(innocentOrderID))[0].should.eql(guiltyOrderID);
+
+        let feeNum = await renExSettlement.DARKNODE_FEES_NUMERATOR();
+        let feeDen = await renExSettlement.DARKNODE_FEES_DENOMINATOR();
+        let weiAmount = eth_decimals.times(ethAmount);
+        let fees = weiAmount / feeDen * feeNum
+
+        // Store the original balances
+        let beforeSlasherBalance = await rewardVault.darknodeBalances(slasher, eth_address);
+        let [beforeGuiltyTokens, beforeGuiltyBalances] = await renExBalances.getBalances(guiltyAddress);
+        let [beforeInnocentTokens, beforeInnocentBalances] = await renExBalances.getBalances(innocentAddress);
+        let beforeGuiltyBalance = beforeGuiltyBalances[beforeGuiltyTokens.indexOf(eth_address)];
+        let beforeInnocentBalance = beforeInnocentBalances[beforeInnocentTokens.indexOf(eth_address)];
+
+        // Slash the fees
+        await renExSettlement.slash(guiltyOrderID, { from: slasher });
+
+        // Check the new balances
+        let afterSlasherBalance = await rewardVault.darknodeBalances(slasher, eth_address);
+        let [afterGuiltyTokens, afterGuiltyBalances] = await renExBalances.getBalances(guiltyAddress);
+        let [afterInnocentTokens, afterInnocentBalances] = await renExBalances.getBalances(innocentAddress);
+        let afterGuiltyBalance = afterGuiltyBalances[afterGuiltyTokens.indexOf(eth_address)];
+        let afterInnocentBalance = afterInnocentBalances[afterInnocentTokens.indexOf(eth_address)];
+
+        // Make sure fees were reallocated correctly
+        let slasherBalanceDiff = afterSlasherBalance - beforeSlasherBalance;
+        let innocentBalanceDiff = afterInnocentBalance - beforeInnocentBalance;
+        let guiltyBalanceDiff = afterGuiltyBalance - beforeGuiltyBalance;
+        // We expect the slasher to have +0.002% fees
```

Comment:
Note that the current fees are 0.2%, or a 0.002 multiplier

Reply:
Removed references to the explicit multiplier amounts 👍